crates/ui/src/modal.rs

```diff
@@ -248,6 +248,9 @@ impl Modal {
     ///
     /// When the overlay is clicked, the modal will be closed.
     pub fn overlay_closable(mut self, overlay_closable: bool) -> Self {
+        if !self.overlay && overlay_closable {
+            eprintln!("Warning: overlay_closable has no effect when overlay is disabled.");
+        }
```

Comment:
Keep 1 PR to 1 thing to help us to review and merge.

This change is not good enough, you can revert it.